### DISCUSSION TITLE
Command conflict problem?

### DISCUSSION THREAD

**Discussion by Smilestate** [socketio/socket.io#4177] [Q&A] [answered]
After connecting to the server, the init command is sent first, and the disconnect command is triggered when the connection is disconnected. Then which instruction will be executed first at this time, whether it is executed after a certain instruction is completed, and the disconnect instruction is executed, or even the init instruction is interrupted, and only the disconnect is executed. Is this the problem I encountered?
```javascrpt
const data = []
socket.on("init", async msg => {
       // 模拟init中执行的代码 
        setInterval(() => {
                trade.putListenKey("binance", listenKey, apiKey, secret)
                console.log(`每30分钟延长listenKe====>${listenKey}`)
          }, 3 * 1000 * 60)
        // 连接成功后 加入对应房间
        console.log(`======当前连接====加入${session_id}房间`)
        data.push("session_id")
});
socket.on("disconnect", async msg => {
        // 监听断开连接
        console.log(`======断开连接====离开${session_id}房间`)
        data = []
});
```
The init command and the disconnect command are triggered at the same time, so does my array have a value?

**Comment by darrachequesne** [comment, marked as answer]
I'm not sure I understand your issue. Could you please provide a complete example reproducing the problem?
